### PR TITLE
Replace list form with List Input in Create a DOS Opportunity journey

### PR DESCRIPTION
**Pull request by domoscargin**
https://trello.com/c/VG4caYdJ/136-3-replace-list-form-with-list-input-component-in-create-a-dos-opportunity-journey

This PR uses the new [content-loader govuk_frontend methods](https://github.com/alphagov/digitalmarketplace-content-loader/pull/90) to use the [List Input component](https://github.com/alphagov/digitalmarketplace-govuk-frontend/tree/master/src/digitalmarketplace/components/list-input).

### CODE REVIEW

File: app/templates/buyers/_base_edit_question_page.html [hunk 1]
```diff
@@ -31,10 +32,14 @@ <h1 class="govuk-heading-l">
...
           {% if question.type != 'multiquestion' %}
             {% if is_govuk_frontend %}
-              {% set govuk_forms = {"govukInput": govukInput} %}
+              {% set govuk_forms = {"govukInput": govukInput, "dmListInput": dmListInput} %}
               {% set form = govuk_frontend_from_question(question, brief, errors) %}
-              {{ govukLabel(form.label) }}
-              {{ question.question_advice }}
+              {%- if form.label %}
+                {{ govukLabel(form.label) }}
+              {% endif -%}
```
Comment (lfdebrux) [inline app/templates/buyers/_base_edit_question_page.html:39]: 👍

**Comment by lfdebrux** [comment]
Testing this locally, I've noticed that the new version is being used for the cultural fit criteria question but not the technical competence criteria question. I'm guessing this is because the latter is a multiquestion; shall we try and crack that problem in this PR? I'm happy to contribute a commit or pair on fixing that.

**Comment by domoscargin** [comment]
> Testing this locally, I've noticed that the new version is being used for the cultural fit criteria question but not the technical competence criteria question. I'm guessing this is because the latter is a multiquestion; shall we try and crack that problem in this PR? I'm happy to contribute a commit or pair on fixing that.

That's a good point. I'd assumed we were handling multiquestions in content loader, but the logic does all appear to be contained here. I've updated the template page. The only issue is headings - we need Content Loader to not set individual question labels/legends as the page heading.

File: app/templates/buyers/_base_edit_question_page.html [hunk 2]
```diff
@@ -53,7 +53,18 @@ <h1 class="govuk-heading-l">
...
+                {% set govuk_forms = {"govukInput": govukInput, "dmListInput": dmListInput} %}
+                {% set form = govuk_frontend_from_question(question, brief, errors) %}
+                {%- if form.label %}
+                  {{ govukLabel(form.label) }}
+                {% endif -%}
+                {% if not form.params.question_advice %}
+                  {{ question.question_advice }}
+                {% endif %}
+                {{ govuk_forms[form.macro_name](form.params) }}
+              {% elif errors and errors[question.id] %}
```
Comment (lfdebrux) [inline app/templates/buyers/_base_edit_question_page.html:67]: Let's encapsulate this logic common to multiquestions and questions in a macro, we can define one at the top of this file https://jinja.palletsprojects.com/en/2.11.x/templates/#macros

**Comment by lfdebrux** [comment]
> Guess we could either pass another arg to `govuk_frontend_from_question` indicating that it's a child, or we could handle multiquestions within Content Loader.

Let's add an optional parameter to `govuk_label` and `govuk_fieldset` `is_page_heading=True`, and also have `from_question` pass on kwargs. We might want to get more sophisticated later but this seems like a good way to start to me.

**Comment by lfdebrux** [comment]
> > Guess we could either pass another arg to `govuk_frontend_from_question` indicating that it's a child, or we could handle multiquestions within Content Loader.
> 
> Let's add an optional parameter to `govuk_label` and `govuk_fieldset` `is_page_heading=True`, and also have `from_question` pass on kwargs. We might want to get more sophisticated later but this seems like a good way to start to me.

Done in [digitalmarketplace-content-loader#7.14.0](https://github.com/alphagov/digitalmarketplace-content-loader/pull/94)

**Review by lfdebrux — APPROVED** [review]
Looks great to me! Well done for getting this working 👍

**Pull request merged**